src/ui/components/LoadingIndicator/index.js

```diff
@@ -0,0 +1,40 @@
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+import Icon from 'ui/components/Icon';
+
+import './LoadingIndicator.scss';
+
+
+export default class LoadingIndicator extends React.Component {
+  static propTypes = {
+    children: PropTypes.node,
+    timeToReset: PropTypes.number,
+  }
+
+  static defaultProps = {
+    children: null,
+    timeToReset: 500,
+  }
+
+  restartAnimation = (event) => {
+    const { timeToReset } = this.props;
+    const iconDOM = ReactDOM.findDOMNode(this.loadingIcon);
+    iconDOM.classList.remove('animating');
```

Comment:
Totally. Sometimes I forget about React state stuff but that's better AND easier to test. 👍